include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,277 @@
+   * All fields are allowed to be empty when writing.
+   *
+   * ### Implementation notes
+   *
+   * This implementation ignores all fields besides id, seq and quality.
+   *
+   */
+class sequence_file_format_sam
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+     //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
+    sequence_file_format_sam() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_format_sam(sequence_file_format_sam const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_format_sam & operator=(sequence_file_format_sam const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_sam(sequence_file_format_sam &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_sam & operator=(sequence_file_format_sam &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "sam" },
+    };
+
+    //!\copydoc sequence_file_input_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                               & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                                  & sequence,
+              id_type                                                                   & id,
+              qual_type                                                                 & qualities)
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        // cache the begin position so we write quals to the same position as seq in seq_qual case
+        size_t sequence_size_before{0};
+        size_t sequence_size_after{0};
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto constexpr is_tab{is_char<'\t'>};
+        auto constexpr is_cntrl{is_char<'\n'>};
+
+        while (is_char<'@'>(*ranges::begin(stream_view)))
```

Comment:
You still have some `ranges::` calls in the files, please replace them with just the function call or `seqan3::` or `std::ranges::`